ground/gcs/src/plugins/uavobjects/uavobjectfield.cpp

```diff
@@ -616,7 +616,11 @@ QString UAVObjectField::toString()
     sout.append ( QString("%1: [ ").arg(name) );
     for (unsigned int n = 0; n < numElements; ++n)
     {
-        sout.append( QString("%1 ").arg(getDouble(n)) );
+        if (type == ENUM) {
+            sout.append( QString("%1 ").arg(getValue(n).toString()) );
+        } else {
+            sout.append( QString("%1 ").arg(getDouble(n)) );
+        }
```

Comment:
I could.  What's here is still clearly broken, e.g. showing UAVO ids in scientific notation.  But I didn't want to fiddle with it too much-- this solved the outright lying that was happening before when "QuadX" became 0